### PR TITLE
OIDC Session Manamement Impl

### PR DESCRIPTION
**Pull request by DMHP**

### CODE REVIEW

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/LogoutEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+public class LogoutEndpoint {
+    private static Log log = LogFactory.getLog(LogoutEndpoint.class);
+
+    @GET
+    @Path("/")
+    @Produces("text/html")
+    public String logout(@Context HttpServletRequest request, @QueryParam("id_token_hint") String id_token_hint,
+                         @Context HttpServletResponse resp, @QueryParam("isLogout") String isLogoutRequest) {
+        try {
+            if (("true").equals(isLogoutRequest)) {
```
Comment (kasungayan) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/LogoutEndpoint.java]: Can use Boolean.parseBoolean();

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java [hunk 2]
```diff
@@ -343,4 +344,8 @@ public static String getHostName() {
         return ServerConfiguration.getInstance().getFirstProperty("HostName");
     }
 
+    public static String getRpInitiatedLogoutUrl(){
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return null;
+    }
+
+    public static void setCallBackUrl(OAuth2AuthorizeReqDTO authzReqDTO) {
+        callBackurl = authzReqDTO.getCallbackUrl();
+    }
+
+    private String getCallBackUrl() {
+               return callBackurl;
+            }
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java]: Formatting issue.

File: components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
+
```
Comment (omindu) [inline components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/session/OIDCSessionMgtEndpoint.java]: Unnecessary new line.

**Comment by madurangasiriwardena** [comment]
Added this feature with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/5

**Pull request closed without merging**